crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -330,6 +330,36 @@ fn single_expression_cycle_initial<'db>(
     Type::Never
 }
 
+/// I tried several attempts of this function:
+/// 1. `infer_expression` similar to `infer_expression_type`. It calls `infer_expression_types`
+///    internally and returns `ExpressionInference`. The problem I faced was that the return type of
+///    this function could not be &`ExpressionInference` because `infer_expression_types` itself is
+///    returning &`ExpressionInference`.
+/// 2. `infer_expression_and_binding`: return both type and `definitely_bound`. In order to overcome
+///    the problem that I cannot have two `salsa::tracked` functions calling each other and both
+///    returning &`ExpressionInference`. It did not help with the cycle because before we get the
+///    type we need to abort if it's not `definitely_bound`.
+/// 3. `infer_expression_if_definitely_bound`: returns `Type::Unknown` if the expression is not `definitely_bound`.
+///    it worked but it did not reduce the number of cycles as much as other way.
+///    This works for the `analyze_cycles_gridout` test case but causes panic for others
```

Comment:
Can we remove this, or make it more concise to mention only what's relevant for posterity?